packages/cookies/src/response-cookies.ts

```diff
@@ -1,88 +1,126 @@
-    const headers = this.headers.getAll('set-cookie')
-    const map = new Map<string, ParsedCookie>()
+  #cache = cached(() => {
+    // @ts-expect-error See https://github.com/whatwg/fetch/issues/973
+    const headers = this.#headers.getAll('set-cookie')
+    const map = new Map<string, Cookie>()
 
     for (const header of headers) {
       const parsed = parseSetCookieString(header)
       if (parsed) {
-        map.set(parsed.name, {
-          value: parsed.value,
-          options: parsed?.attributes,
-        })
+        map.set(parsed.name, parsed)
       }
     }
 
     return map
   })
 
-  private parsed() {
-    const allCookies = this.headers.get('set-cookie')
-    return this.cache(allCookies)
+  #parsed() {
+    const allCookies = this.#headers.get('set-cookie')
+    return this.#cache(allCookies)
   }
 
-  set(key: string, value: string, options?: Options): this {
-    const map = this.parsed()
-    map.set(key, { value, options: normalizeCookieOptions(options || {}) })
-    replace(map, this.headers)
+  /**
+   * {@link https://wicg.github.io/cookie-store/#CookieStore-get CookieStore#get} without the Promise.
+   */
+  get(...args: [key: string] | [options: Cookie]): Cookie | undefined {
+    const key = typeof args[0] === 'string' ? args[0] : args[0].name
+    return this.#parsed().get(key)
+  }
+  /**
+   * {@link https://wicg.github.io/cookie-store/#CookieStore-getAll CookieStore#getAll} without the Promise.
+   */
+  getAll(...args: [key: string] | [options: Cookie] | [undefined]): Cookie[] {
+    const all = Array.from(this.#parsed().values())
+    if (!args.length) {
+      return all
+    }
+
+    const key = typeof args[0] === 'string' ? args[0] : args[0]?.name
+    return all.filter((c) => c.name === key)
+  }
+
+  /**
+   * {@link https://wicg.github.io/cookie-store/#CookieStore-set CookieStore#set} without the Promise.
+   */
+  set(
+    ...args:
+      | [key: string, value: string, cookie?: Partial<Cookie>]
+      | [options: Cookie]
+  ): this {
+    const [name, value, cookie] =
+      args.length === 1 ? [args[0].name, args[0].value, args[0]] : args
+    const map = this.#parsed()
+    map.set(name, normalizeCookie({ name, value, ...cookie }))
+    replace(map, this.#headers)
 
     return this
   }
 
-  delete(key: string): this {
-    return this.set(key, '', { expires: new Date(0) })
+  /**
+   * {@link https://wicg.github.io/cookie-store/#CookieStore-delete CookieStore#delete} without the Promise.
+   */
+  delete(...args: [key: string] | [options: Cookie]): this {
+    const name = typeof args[0] === 'string' ? args[0] : args[0].name
+    return this.set({ name, value: '', expires: new Date(0) })
   }
 
-  get(key: string): string | undefined {
-    return this.getWithOptions(key)?.value
+  // Non-spec
+
+  /**
+   * Uses {@link ResponseCookies.get} to return only the cookie value.
+   */
+  getValue(...args: [key: string] | [options: Cookie]): string | undefined {
+    return this.get(...args)?.value
```

Comment:
Arguably, this method should be unnecessary:
`.getValue("cookie")` is the same as
`.get("cookie")?.value`

Breaking spec to spare 2 characters might not be worth it.